lib/improver/tests/test_nbhood_neighbourhoodprocessing.py

```diff
@@ -0,0 +1,381 @@
+    for time_index, lat_index, lon_index in zero_point_indices:
+        data[time_index][lat_index][lon_index] = 0
+    cube = Cube(data, standard_name="precipitation_amount",
+                units="kg m^-2 s^-1")
+    tunit = Unit("hours since 1970-01-01 00:00:00", "gregorian")
+    time_points = [402192.5 + _ for _ in range(num_time_points)]
+    cube.add_aux_coord(
+        AuxCoord(time_points, "time", units=tunit), 0)
+    cube.add_dim_coord(
+        DimCoord(np.linspace(0.0, float(num_grid_points - 1),
+                             num_grid_points),
+                 'latitude',
+                 units='degrees'),
+        1
+    )
+    cube.add_dim_coord(
+        DimCoord(np.linspace(0.0, float(num_grid_points - 1),
+                             num_grid_points),
+                 'longitude',
+                 units='degrees'),
+        2
+    )
+    return cube
+
+
+class Test__init__(IrisTest):
+
+    def test_radii_varying_with_lead_time_mismatch(self):
+        """
+        Test that the desired error message is raised, if there is a mismatch
+        between the number of radii and the number of lead times.
+        """
+        radii_in_km = [10, 20, 30]
+        lead_times = [2, 3]
+        msg = "There is a mismatch in the number of radii"
+        with self.assertRaisesRegexp(ValueError, msg):
+            kernel_method = 'circular'
+            NBHood(kernel_method, radii_in_km, lead_times=lead_times)
+
+    def test_kernel_method_does_not_exist(self):
+        """
+        Test that desired error message is raised, if the kernel method
+        does not exist.
+        """
+        kernel_method = 'nonsense'
+        radii_in_km = 10
+        msg = 'The requested kernel method: '
+        with self.assertRaisesRegexp(AttributeError, msg):
+            NBHood(kernel_method, radii_in_km)
+
+
+class Test_process(IrisTest):
+
+    """Tests for the process method of BasicNeighbourhoodProcessing."""
+
+    RADIUS_IN_KM = 6.3  # Gives 3 grid cells worth.
+
+    def test_basic(self):
+        """Test that the plugin returns an iris.cube.Cube."""
+        cube = set_up_cube()
+        kernel_method = "circular"
+        plugin = NBHood(kernel_method, self.RADIUS_IN_KM)
+        result = plugin.process(cube)
+        self.assertIsInstance(result, Cube)
+
+    def test_single_point_nan(self):
+        """Test behaviour for a single NaN grid cell."""
+        cube = set_up_cube()
+        cube.data[0][0][6][7] = np.NAN
+        msg = "NaN detected in input cube data"
+        with self.assertRaisesRegexp(ValueError, msg):
+            kernel_method = "circular"
+            NBHood(kernel_method, self.RADIUS_IN_KM).process(cube)
+
+    def test_fail_multiple_realisations(self):
+        """Test failing when the array has a realisation dimension."""
+        data = np.ones((14, 1, 16, 16))
+        data[0][0][7][7] = 0.0
+
+        cube = Cube(data, standard_name="precipitation_amount",
```

Comment:
Good idea. I'm not sure why this unit test was doing so much work itself, but we've standardised this now.